sources/packages/backend/apps/api/src/route-controllers/announcements/announcements.controller.ts

```diff
@@ -0,0 +1,20 @@
+import { Controller, Get } from "@nestjs/common";
+import BaseController from "../BaseController";
+import { ApiTags } from "@nestjs/swagger";
+import { Announcements } from "@sims/sims-db";
+import { AnnouncementsService } from "../../services";
+import { Public } from "../../auth/decorators/public.decorator";
+
+@Controller("announcements")
+@ApiTags("announcements")
+export class AnnouncementsController extends BaseController {
+  constructor(private readonly announcementsService: AnnouncementsService) {
+    super();
+  }
+
+  @Public()
+  @Get()
+  async getAnnouncements(): Promise<Announcements[]> {
+    return this.announcementsService.getAnnouncements();
```

Comment:
Should the API return all the system announcements for all the clients ? 

I meant, when this is called from student web, it will still return the announcements that are intended for institutions as well in the API. 

Are we considering API call using a student token receiving institution announcements to be safe?

Thoughts @andrewsignori-aot @andrepestana-aot @guru-aot @sh16011993 @bidyashish ?

Reply:
First, the AC states below where the API must receive the target to allow filtering the messages. So returning all messages was not the expectation.
```
Create a Vue component.
- Property to receive the array of strings (targets) to be retrieved.
```

Reply:
Second, we may have wrongly planned the controller to be generic. It must be client-specific to allow only the specific targets to be provided for the specific clients.

Reply:
Split into student and institution controllers